### PR TITLE
fix(dropdown): update callback function call on selected value in menu variant

### PR DESCRIPTION
**Pull request by samyak3009**
### What does this implement/fix? Explain your changes.
we can select the option again which is already selected and callback function will also called everytime we select the already selected value.
...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

**Review by anuradha9712 — CHANGES_REQUESTED** [review]
Please add a test case for both the scenarios when `menu={true}` & `menu={false}`

**Comment by satyamyadav** [comment]
@samyak3009  Can you please update/add the test case.

**Review by anuradha9712 — CHANGES_REQUESTED** [review]
Please add a valid test case for Controlled dropdown as well and check for both `onChange` & `onUpdate` callback functions to be triggered correctly.

**Review by anuradha9712 — CHANGES_REQUESTED** [review]
Please change the commit message

File: core/components/atoms/dropdown/__tests__/Dropdown.test.tsx [hunk 1]
```diff
@@ -469,6 +469,79 @@ describe('Dropdown component with search', () => {
...
+    await waitFor(() => {
+      optionList = getAllByTestId('DesignSystem-DropdownOption--DEFAULT');
+    });
+    fireEvent.click(optionList[1]);
+    fireEvent.click(optionList[1]);
+    expect(FunctionValue).toHaveBeenCalledTimes(2);
+  });
+  it('check for selecting already selected value in dropdown', async () => {
+    jest.resetAllMocks();
+    const { getByTestId, getAllByTestId } = render(<Dropdown options={dropdownOptions} onChange={FunctionValue} />);
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/__tests__/Dropdown.test.tsx]: Please add a new line after `it` block

**Review by anuradha9712 — APPROVED** [review]

**Pull request merged**